TestModels/dafny-dependencies/StandardLibrary/runtimes/python/src/standard_library/extern/UTF8.py

```diff
@@ -0,0 +1,86 @@
+from standard_library.internal_generated_dafny.UTF8 import *
+import standard_library.internal_generated_dafny.UTF8
+import _dafny
+import struct
+
+'''
+Extern UTF8 encode and decode methods.
+
+Note:
+Python's `.encode('utf-8')` does not handle surrogates.
+These methods are expected to handle surrogates (e.g. "\uD808\uDC00").
+To work around this, we encode Dafny strings into UTF-16-LE (little endian)
+and decode them before encoding into UTF-8 (`_strict_tostring`).
+To decode, we reverse the encode step. (`_reverse_strict_tostring`)
```

Comment:
Question: Tell me about the performance characteristics of this implementation.
It appears that we parse every character three times.

For Encoding:
1. Encode to UTF-16-LE
2. Decode to ?
3. Encode to UTF-8

For Decode:
1. Decode from UTF-8

Is this correct?

Reply:
We are actually doing 3 encodings on decode as well (reverse of encode).

Reply:
We do not need to block development on this.
But we should not ship a product with this deficiency.

I do not know how often we convert strings,
but I imagine it's often.

Reply:
This might be a good time to at least discuss the eventual path to using `--unicode-char:true` in Dafny. A big part of supporting that mode is avoiding exactly this annoying edge case.